commcare/src/main/resources/velocity.templates/task-data-provider.vm

```diff
@@ -163,7 +163,7 @@
                                     "displayName" : "commcare.case.field.caseName",
                                     "fieldKey" : "caseName"
                                 },
-                                #foreach($caseField in $entry.value)
+                                #foreach($caseField in $entry.getApplicationName())
```

Comment:
From what I recall, this loop was used to generate data source fields for cases (each individual case can have different field names). There's an obvious problem here, since you cannot iterate over a String (getApplicationName()). This should call something that will return a collection of the given case fields.